### PR TITLE
added ES5 support

### PR DESCRIPTION
**Pull request by bySabi**
Here we go 
 Is a big PR with some changes.
- update packages
- add regenerator for transform to es5
- add package.json scripts for `build` and `prepare` npm before `publish`
- update Doc

I hope it´s all fine to you

### CODE REVIEW

**Comment by bySabi** [comment]
I left es5 files on repo so the module can be installed from github

File: package.json [hunk 1]
```diff
@@ -11,23 +11,30 @@
...
-    "chai": "3.5.0",
-    "istanbul": "0.4.2",
-    "mocha": "2.4.5"
+    "regenerator": "~0.9.7",
+    "chai": "4.x.x",
+    "istanbul": "~0.4.5",
+    "mocha": "3.x.x"
   },
   "engines": {
     "node": ">=4.2"
```
Comment (alvaropinot) [inline package.json]: maybe we could use an older version of node with this PR :)

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "circular-iterator",
-  "version": "1.1.5",
+  "version": "1.2.0",
```
Comment (alvaropinot) [inline package.json]: ❤️

**Review by alvaropinot — APPROVED** [review]
Thank you for this! Looks really good. Maybe add your self as a contributor at the `package.json` 😄

**Comment by alvaropinot** [comment]
> I left es5 files on repo so the module can be installed from github

@bySabi it's a good idea, as using npm post install scripts will make the installing non deterministic.

File: package.json (hunk 1 above)
Comment (bySabi) [reply package.json]: node 4.0 or more older ?

Comment (alvaropinot) [reply package.json]: I think it could be any version with then ES5 transpilation.

Comment (bySabi) [reply package.json]: ah! right!
you can put the limit on  "node": ">= 0.6", is the minimal support of regenerator runtime

Comment (bySabi) [reply package.json]: We'll need additional files and setup for test ES5 on travis.

**Comment by bySabi** [comment]
Add a new commit with node >= 0.6  support.

Code coverage will need a little tweeks, we don´t need to cover regenerator transformed code.

**Review by alvaropinot — APPROVED** [review]
cool :)

**Pull request merged**